app/src/main/java/org/mozilla/focus/tabs/TabViewClient.java

```diff
@@ -0,0 +1,32 @@
+/* -*- Mode: Java; c-basic-offset: 4; tab-width: 4; indent-tabs-mode: nil; -*-
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.focus.tabs;
+
+/**
+ * An abstract layer of @see{android.webkit.WebViewClient}
+ */
+public class TabViewClient {
+
+    public void onPageStarted(String url) {
+    }
+
+    public void onPageFinished(boolean isSecure) {
+    }
+
+    public void onURLChanged(String url) {
+    }
+
+    /**
+     * Return true if the URL was handled, false if we should continue loading the current URL.
+     */
+    public boolean handleExternalUrl(String url) {
+        return false;
+    }
+
+
+    public void updateFailingUrl(String url, boolean updateFromError) {
```

Comment:
This one is a bit cumbersome as a solution to "Not insert failing urls into histories" Maybe we'll have a better way to solve this when we decided to update our history update timing.